orchestrator/careplancontributor/ehr/kafka_client.go

```diff
@@ -0,0 +1,169 @@
+//
+// Returns:
+//   - KafkaClient: An implementation of the KafkaClient interface.
+//   - error: An error if the Kafka producer could not be created.
+func NewClient(config KafkaConfig) (KafkaClient, error) {
+
+	var kafkaClient KafkaClient
+	if config.Enabled {
+		endpoint := config.Endpoint
+		log.Infof("KafkaClientImpl, connecting to %s, Mechanism: %s, protocol: %s, username: %s, password: %s",
+			endpoint,
+			config.Sasl.Mechanism,
+			config.Security.Protocol,
+			config.Sasl.Username,
+			config.Sasl.Password)
+		producer, err := kafka.NewProducer(&kafka.ConfigMap{
+			"bootstrap.servers": endpoint,
+			"sasl.mechanisms":   config.Sasl.Mechanism,
+			"sasl.username":     config.Sasl.Username,
+			"sasl.password":     config.Sasl.Password,
+			"security.protocol": config.Security.Protocol,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Delivery report handler for produced messages
+		go func() {
+			log.Infof("Kafka func started")
+			for e := range producer.Events() {
+				log.Infof("Kafka event received: %v", e.String())
+				switch ev := e.(type) {
+				case *kafka.Message:
+					if ev.TopicPartition.Error != nil {
+						log.Infof("Kafka delivery failed: %v\n", ev.TopicPartition)
+					} else {
+						log.Errorf("Kafka delivered message to %v\n", ev.TopicPartition)
+					}
+				}
+			}
+			log.Infof("Kafka func ended")
+		}()
+
+		kafkaClient = &KafkaClientImpl{
+			topic:    config.Topic,
+			producer: producer,
+		}
+	} else {
+		kafkaClient = &NoopClientImpl{}
+	}
+	return kafkaClient, nil
+}
+
+// SubmitMessage submits a message to the Kafka topic associated with the KafkaClientImpl.
+// It produces a message with the given key and value to the Kafka producer.
+//
+// Parameters:
+//   - key: The key of the message to be submitted.
+//   - value: The value of the message to be submitted.
+//
+// Returns:
+//   - error: An error if the message could not be produced.
+func (k *KafkaClientImpl) SubmitMessage(key string, value string) error {
+	log.Infof("SubmitMessage, submitting key %s", key)
+	err := k.producer.Produce(&kafka.Message{
+		TopicPartition: kafka.TopicPartition{
+			Topic:     &k.topic,
+			Partition: kafka.PartitionAny,
+		},
+		Key:   []byte(key),
+		Value: []byte(value),
+	}, nil)
+	if err != nil {
+		return fmt.Errorf("failed to write message: %w", err)
+	}
+	k.producer.Flush(10 * 1000)
+	log.Infof("SubmitMessage, submitted key %s", key)
+	return nil
+}
+
+// SubmitMessage writes a message to a file in the /tmp directory.
+// The filename is derived from the key by replacing colons with underscores
+// and appending the .json extension.
+//
+// Parameters:
+//   - key: The key of the message, used to generate the filename.
+//   - value: The value of the message, written to the file.
+//
+// Returns:
+//   - error: An error if the file could not be written.
+func (k *NoopClientImpl) SubmitMessage(key string, value string) error {
+	name := "/tmp/" + strings.ReplaceAll(key, ":", "_") + ".json"
```

Comment:
this is not noop and feels like very bad practice for production, to me

Reply:
Reworked the code into `DebugClient` that writes the messages to the /tmp dir and the `NoopClient` that does nothing.